src/main/java/fi/aalto/cs/apluscourses/presentation/CourseItemViewModel.java

```diff
@@ -0,0 +1,35 @@
+package fi.aalto.cs.apluscourses.presentation;
+
+import org.jetbrains.annotations.NotNull;
+
+public class CourseItemViewModel {
+
+  private final String name;
+  private final String semester;
+  private final String url;
+
+  /**
+   * Construct a view model with the given parameters.
+   */
+  public CourseItemViewModel(@NotNull String name, @NotNull String semester, @NotNull String url) {
```

Comment:
is the intent to have the class immutable or it just happens to have some characteristics of such?

Reply:
I don't see any reason why this wouldn't be immutable. None of the fields are expected to change.